src/settings/SettingsAddress.js

```diff
@@ -58,18 +57,7 @@ export default class SettingsAddress extends React.Component {
 			address_postcode: state.postcode,
 			phone: state.tel,
 		};
-		LiveData.api.setAccountSettings(req).then(
-			response => {
-				if (response.set_settings === 1) {
-					this.actions.updateSettingFields(req);
-				} else {
-					SettingsAddress.handleUpdateError(response);
-				}
-			},
-			response => {
-				SettingsAddress.handleUpdateError(response);
-			}
-		);
+		this.props.actions.attemptUpdateSettings(req, SettingsAddress.handleUpdateError);
```

Comment:
You are thinking about actions incorrectly. They have too broad name, but narrower purpose.
An action is a piece of data that changes the state.
It should not fail, the action itself.
And should not return data.
LiveData is handling the requesting and receiving in two places, and it seems to not matter, for most or all cases. If you get a tick, you store it, and don't care who actually requested it.